catalog/transformer/catalog-transformer-streaming-impl/src/main/java/org/codice/ddf/transformer/xml/streaming/impl/Gml3ToWktImpl.java

```diff
@@ -106,13 +115,86 @@ public Object parseXml(InputStream xml) throws ValidationException {
     }
   }
 
-  private Geometry convertCRS(Geometry geometry) throws ValidationException, TransformException {
-    return JTS.transform(geometry, getLatLonTransform());
+  /**
+   * Iterate over the geometry collection and convert the non-standard LinearRing to LineString
+   *
+   * @param geometryCollection the geometryCollection to be iterated
+   * @return an edited geometryCollection with LineString in place of LinearRing
+   */
+  private GeometryCollection standardizeGeometryCollection(GeometryCollection geometryCollection) {
+    ArrayList<Geometry> geometries = new ArrayList<>();
+
+    for (int i = 0; i < geometryCollection.getNumGeometries(); i++) {
+      Geometry geometry = standardizeGeometry(geometryCollection.getGeometryN(i));
+      geometries.add(geometry);
+    }
+
+    return GEOMETRY_FACTORY.createGeometryCollection(geometries.toArray(new Geometry[0]));
+  }
+
+  /**
+   * Convert the non-standard LinearRing geometry to a LineString geometry
+   *
+   * @param geometry LinearRing to be converted
+   * @return new LineString geometry
+   */
+  private Geometry linearRingToLineString(Geometry geometry) {
+    return GEOMETRY_FACTORY.createLineString(((LinearRing) geometry).getCoordinateSequence());
+  }
+
+  /**
+   * Applies the math transformation to change Coordinate Reference Systems
+   *
+   * @param geometry Geometry to be transformed
+   * @return a Geometry with the transformation applied
+   * @throws ValidationException
+   * @throws TransformException
+   */
+  private Geometry applyMathTransform(Geometry geometry)
+      throws ValidationException, TransformException {
+    CoordinateReferenceSystem geometryCrs = getGeometryCrs(geometry);
+    return JTS.transform(geometry, getLatLonTransform(geometryCrs));
+  }
+
+  /**
+   * Standardizes the geometry by converting any instances of LinearRing to LineString.
+   *
+   * @param geometry the Geometry that may contain LinearRing
+   * @return a new Geometry with LineStrings in place of LinearRings
+   * @throws ValidationException
+   */
+  private Geometry standardizeGeometry(Geometry geometry) {
+    if (geometry instanceof GeometryCollection && !(geometry instanceof MultiPolygon)) {
+      return standardizeGeometryCollection((GeometryCollection) geometry);
+    }
+
+    if (geometry instanceof LinearRing) {
+      return linearRingToLineString(geometry);
+    }
+
+    return geometry;
+  }
+
+  /**
+   * Get the CRS from the provided geometry, defaulting to WGS84 if CRS not found in geometry.
+   *
+   * @param geometry a geometry used to determine CRS
+   * @return Extracted CRS or default
+   */
+  private CoordinateReferenceSystem getGeometryCrs(Geometry geometry) {
+    CoordinateReferenceSystem crs = (CoordinateReferenceSystem) geometry.getUserData();
```

Comment:
❓ Is it ever possible for the user data to not be `CoordinateReferenceSystem`?

Reply:
I don't think so but I changed it to check instanceof just to be safe.